src/org/zaproxy/zap/extension/alert/PopupMenuItemAlert.java

```diff
@@ -0,0 +1,129 @@
+
+    public PopupMenuItemAlert(String label) {
+        this(label, false);
+    }
+
+	public PopupMenuItemAlert(String label, boolean multiSelect) {
+		super(label);
+        this.multiSelect = multiSelect;
+        addActionListener(new PerformActionsActionListener());
+        this.extAlert = Control.getSingleton().getExtensionLoader().getExtension(ExtensionAlert.class);
+	}
+
+    public final boolean isMultiSelect() {
+        return multiSelect;
+    }
+
+    private HashSet<Alert> getAlertNodes() {
+    	TreePath[] paths = this.extAlert.getAlertPanel().getTreeAlert().getSelectionPaths();
+        HashSet<Alert> alertNodes = new HashSet<Alert>();
+        for(int i = 0; i < paths.length; i++ ) {
+            DefaultMutableTreeNode alertNode = (DefaultMutableTreeNode)paths[i].getLastPathComponent();
+            System.out.println((Alert)alertNode.getUserObject());
+            if(alertNode.getChildCount() == 0) {
+                alertNodes.add(((Alert)alertNode.getUserObject()));
+                continue;
+            }
+            for(int j = 0; j < alertNode.getChildCount(); j++ ) {
+                DefaultMutableTreeNode node = (DefaultMutableTreeNode)alertNode.getChildAt(j);
+                alertNodes.add(((Alert)node.getUserObject()));
+            }
+        }
+        return alertNodes;
+    }
+
+    protected abstract void performAction(Alert alert);
+
+    protected void performActions(HashSet<Alert> alerts) {
+        for(Alert alert: alerts) {
+            performAction(alert);
+        }
+    }
+
+    protected boolean isButtonEnabledForNumberOfSelectedAlerts(int count) {
+        if(count == 0 ) {
+            return false;
+        } else if(!isMultiSelect() && count>1 ) {
+            return false;
+        }
+        return true;
+    }
+
+    @Override
+    public boolean isEnableForComponent(Component invoker) {
+        TreePath[] paths = this.extAlert.getAlertPanel().getTreeAlert().getSelectionPaths();
+        for(int i = 0; i < paths.length; i++ ){
+            DefaultMutableTreeNode alertNode = (DefaultMutableTreeNode)paths[i].getLastPathComponent();
+            if(alertNode.getUserObject() == null) {
+                return false;
+            }
+        }
+        if(this.extAlert == null) {
+            return false;
+        }
+        if ("treeAlert".equals(invoker.getName())) {
+            int count = this.extAlert.getAlertPanel().getTreeAlert().getSelectionCount();
+            setEnabled(isButtonEnabledForNumberOfSelectedAlerts(count));
+            return true;
+        }
+        return false;
+    }
+
+    private class PerformActionsActionListener implements ActionListener {
+
+        @Override
+        public void actionPerformed(ActionEvent evt) {
+            try {
+                HashSet<Alert> alerts = getAlertNodes();
+                performActions(alerts);
+            } catch (Exception e) {
+                log.debug(e.getMessage(),e);
```

Comment:
This should be logged as error, if an exception gets this far it's most likely a bug in the custom menu items.